libs/jsonschema/utils.go

```diff
@@ -111,3 +112,32 @@ func FromString(s string, T Type) (any, error) {
 	}
 	return v, err
 }
+
+func ValidatePatternMatch(name string, value any, propertySchema *Schema) error {
+	if propertySchema.Pattern == "" {
+		// Return early if no pattern is specified
+		return nil
+	}
+
+	// Expect type of value to be a string
+	stringValue, ok := value.(string)
+	if !ok {
+		return fmt.Errorf("invalid value for %s: %v. Expected a value of type string", name, value)
+	}
```

Comment:
Isn't this the responsibility of another checker?

Reply:
Yeah it is. This is present to avoid panicing if the typecast ever fails.